src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -43,15 +44,26 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 		public System.Type BaseType { get; private set; }
 		public System.Type[] Interfaces { get; private set; }
```

Comment:
As we are allowed to break everything in NH5, Please make this `IReadOnlyCollection<System.Type>` without getter.